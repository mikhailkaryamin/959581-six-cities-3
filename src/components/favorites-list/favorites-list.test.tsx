import * as React from 'react';
import * as renderer from 'react-test-renderer';
import configureStore from "redux-mock-store";
import {
  Provider
} from "react-redux";
import {
  BrowserRouter
} from 'react-router-dom';
import {
  Store
} from "../../mocks/initialStateTest";
import {
  OFFERS,
  LOCATIONS,
} from '../../mocks/constsMockTest';
import FavoritesList from './favorites-list';

const mockStore = configureStore([]);
const onCardHover = jest.fn();
const onCardLeave = jest.fn();

test(`Favorites list render correctly`, () => {
  const store = mockStore(Store.WITH_AUTH);
  const tree = renderer
    .create(
        <Provider
          store={store}
        >
          <BrowserRouter>
            <FavoritesList
              currentCity={OFFERS}
              favorites={OFFERS}
              favoritesLocations={LOCATIONS}
              onCardHover={onCardHover}
              onCardLeave={onCardLeave}
            />
          </BrowserRouter>
        </Provider>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
