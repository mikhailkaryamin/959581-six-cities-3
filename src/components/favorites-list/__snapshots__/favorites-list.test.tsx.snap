// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites list render correctly 1`] = `
<section
  className="favorites"
>
  <h1
    className="favorites__title"
  >
    Saved listing
  </h1>
  <ul
    className="favorites__list"
  >
    <li
      className="favorites__locations-items"
    >
      <div
        className="favorites__locations locations "
      >
        <div
          className="locations__item"
        >
          <a
            className="locations__item-link"
            href="#"
          >
            <span>
              Paris
            </span>
          </a>
        </div>
      </div>
      <div
        className="favorites__places"
      />
    </li>
    <li
      className="favorites__locations-items"
    >
      <div
        className="favorites__locations locations "
      >
        <div
          className="locations__item"
        >
          <a
            className="locations__item-link"
            href="#"
          >
            <span>
              Cologne
            </span>
          </a>
        </div>
      </div>
      <div
        className="favorites__places"
      />
    </li>
    <li
      className="favorites__locations-items"
    >
      <div
        className="favorites__locations locations "
      >
        <div
          className="locations__item"
        >
          <a
            className="locations__item-link"
            href="#"
          >
            <span>
              Brussels
            </span>
          </a>
        </div>
      </div>
      <div
        className="favorites__places"
      />
    </li>
    <li
      className="favorites__locations-items"
    >
      <div
        className="favorites__locations locations "
      >
        <div
          className="locations__item"
        >
          <a
            className="locations__item-link"
            href="#"
          >
            <span>
              Amsterdam
            </span>
          </a>
        </div>
      </div>
      <div
        className="favorites__places"
      >
        <article
          className="favorites__place-card place-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="cities__image-wrapper place-card__image-wrapper"
          >
            <a
              href="#"
            >
              <img
                alt="Place image"
                className="place-card__image"
                height={110}
                src="img/1.png"
                width={150}
              />
            </a>
          </div>
          <div
            className="place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  120
                </b>
                <span
                  className="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                className="favorites__bookmark-button button "
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height={19}
                  width={18}
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "100%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
            >
              <a
                href="/offer/1"
                onClick={[Function]}
              >
                Beautiful & luxurious studio at great location
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              apartment
            </p>
          </div>
        </article>
        <article
          className="favorites__place-card place-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="cities__image-wrapper place-card__image-wrapper"
          >
            <a
              href="#"
            >
              <img
                alt="Place image"
                className="place-card__image"
                height={110}
                src="img/1.png"
                width={150}
              />
            </a>
          </div>
          <div
            className="place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  120
                </b>
                <span
                  className="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                className="favorites__bookmark-button button "
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height={19}
                  width={18}
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "100%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
            >
              <a
                href="/offer/2"
                onClick={[Function]}
              >
                Beautiful & luxurious studio at great location
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              apartment
            </p>
          </div>
        </article>
        <article
          className="favorites__place-card place-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="cities__image-wrapper place-card__image-wrapper"
          >
            <a
              href="#"
            >
              <img
                alt="Place image"
                className="place-card__image"
                height={110}
                src="img/1.png"
                width={150}
              />
            </a>
          </div>
          <div
            className="place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  120
                </b>
                <span
                  className="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                className="favorites__bookmark-button button "
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height={19}
                  width={18}
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "100%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
            >
              <a
                href="/offer/3"
                onClick={[Function]}
              >
                Beautiful & luxurious studio at great location
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              apartment
            </p>
          </div>
        </article>
        <article
          className="favorites__place-card place-card"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="cities__image-wrapper place-card__image-wrapper"
          >
            <a
              href="#"
            >
              <img
                alt="Place image"
                className="place-card__image"
                height={110}
                src="img/1.png"
                width={150}
              />
            </a>
          </div>
          <div
            className="place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  120
                </b>
                <span
                  className="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                className="favorites__bookmark-button button "
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height={19}
                  width={18}
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "100%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
            >
              <a
                href="/offer/4"
                onClick={[Function]}
              >
                Beautiful & luxurious studio at great location
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              apartment
            </p>
          </div>
        </article>
      </div>
    </li>
    <li
      className="favorites__locations-items"
    >
      <div
        className="favorites__locations locations "
      >
        <div
          className="locations__item"
        >
          <a
            className="locations__item-link"
            href="#"
          >
            <span>
              Hamburg
            </span>
          </a>
        </div>
      </div>
      <div
        className="favorites__places"
      />
    </li>
    <li
      className="favorites__locations-items"
    >
      <div
        className="favorites__locations locations "
      >
        <div
          className="locations__item"
        >
          <a
            className="locations__item-link"
            href="#"
          >
            <span>
              Dusseldorf
            </span>
          </a>
        </div>
      </div>
      <div
        className="favorites__places"
      />
    </li>
  </ul>
</section>
`;
